### ISSUE TITLE
Design and implement email verification system

### ISSUE THREAD

**Issue by buckyroberts** [thenewboston-developers/Website-API#42]
Design and implement an email system using AWS SES to allow us to send verification emails to the user.

AWS SES - https://aws.amazon.com/ses/
Frontend ticket - https://github.com/thenewboston-developers/Website/issues/1268

# Basic flow

Note that after clicking the email verification link the users should be taken to their **Profile** (not a **Dashboard**)

<img width="728" alt="Screen Shot 2021-02-07 at 8 43 45 PM" src="https://user-images.githubusercontent.com/8547538/107176793-3541be80-6985-11eb-97b6-efbf76703cdb.png">

# Current

Currently, users can create an account however afterwards they are unable to log in. 

<img width="1126" alt="Screen Shot 2021-02-07 at 8 38 54 PM" src="https://user-images.githubusercontent.com/8547538/107176597-afbe0e80-6984-11eb-9723-c7be5fc2d0a6.png">

There is an `is_email_verified` on the User model that is false by default. This will be the field that is set to true when their email is verified.

<img width="340" alt="Screen Shot 2021-02-07 at 8 45 04 PM" src="https://user-images.githubusercontent.com/8547538/107176978-8f428400-6985-11eb-8943-53ab606fd316.png">

# Other

Q) What if a user clicks the verification link again after already verifying?
A) Do some quick research on the most common practice and follow that pattern

**Comment by thesanjeevsharma** [comment]
> What if a user clicks the verification link again after already verifying?
Well, since the `is_email_verified` is true already. It won't have any effect. They'll be redirected to the profile page.

When I built something like this. I used JWT token in the verification link. So, there would be a 2 days time to verify the email. After that, they would have to generate another link. In our case, I think it has no expiry so we're good.
